taxonomy/api.py

```diff
@@ -0,0 +1,245 @@
+"""APIs for lore taxonomy application"""
+
+from django.db import transaction
+
+from taxonomy.models import (
+    Vocabulary,
+    Term,
+)
+from learningresources.models import (
+    LearningResource,
+    LearningResourceType,
+    Repository,
+)
+
+
+# pylint: disable=no-member
+def get_vocabulary(vocabulary_id):
+    """
+    Lookup vocabulary given its id
+
+    Args:
+        vocabulary_id (int): Vocabulary id
+
+    Returns:
+        Vocabulary: The vocabulary from the database
+    """
+    return Vocabulary.objects.get(id=vocabulary_id)
+
+
+# pylint: disable=too-many-arguments
+@transaction.atomic
+def create_vocabulary(
+        repository_id, name, description, required, vocabulary_type, weight
+):
+    """
+    Create a Vocabulary and save it in database
+
+    Args:
+        repository_id (int): Repository id
+        name (unicode): Name of Vocabulary
+        description (unicode): Description of Vocabulary
+        required (bool): Is Vocabulary required?
+        vocabulary_type (unicode): Vocabulary type
+        weight (int): Weight
+
+    Returns:
+        Vocabulary: The Vocabulary which was just added to the database
+
+    """
+
+    if vocabulary_type != "m" and vocabulary_type != "f":
+        raise TaxonomyAPIException("vocabulary_type must be either"
+                                   " 'm' (managed) or 'f' (free tagging)")
+
+    repository = Repository.objects.get(id=repository_id)
+
+    return Vocabulary.objects.create(
+        repository=repository,
+        name=name,
+        description=description,
+        required=required,
+        vocabulary_type=vocabulary_type,
+        weight=weight,
+    )
+
+
+def delete_vocabulary(vocabulary_id):
+    """
+    Delete Vocabulary with given id
+
+    Args:
+        vocabulary_id (int): Vocabulary id
+    """
+    Vocabulary.objects.filter(id=vocabulary_id).delete()
```

Comment:
This is dangerous, because it could cascade delete other records.

Please see the [on_delete](https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.ForeignKey.on_delete) documentation and update the model as appropriate.